Bazel: fatal error: 'runtime/cpp/emboss_cpp_util.h' file not found

Clang seems unable to find the cpp_utils headers when the emboss_cc_library rule is used.

```
bazel-out/k8-fastbuild/bin/external/pigweed/pw_bluetooth/public/pw_bluetooth/hci_common.emb.h:12:10: fatal error: 'runtime/cpp/emboss_cpp_util.h' file not found
   12 | #include "runtime/cpp/emboss_cpp_util.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/build_defs.bzl
+++ b/build_defs.bzl
@@ -175,8 +175,7 @@ def _cc_emboss_aspect_impl(target, ctx):
         transitive = [
                          dep[EmbossCcHeaderInfo].transitive_headers
                          for dep in ctx.rule.attr.deps
-                     ] +
-                     [runtime_cc_info.compilation_context.headers],
+                     ],
     )
     (cc_compilation_context, cc_compilation_outputs) = cc_common.compile(
         name = ctx.label.name,
@@ -185,6 +184,7 @@ def _cc_emboss_aspect_impl(target, ctx):
         cc_toolchain = cc_toolchain,
         public_hdrs = headers,
         private_hdrs = transitive_headers.to_list(),
+        compilation_contexts = [runtime_cc_info.compilation_context],
     )
     return [
         CcInfo(compilation_context = cc_compilation_context),

